### PR TITLE
[WIP][python-exp] Force camelization of imports

### PR DESCRIPTION
**Pull request by timgclark**
* Forces formatting of import classnames to handle edge case schemas

If merged this will resolve:
https://github.com/OpenAPITools/openapi-generator/issues/7187

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [ ] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java [hunk 1]
```diff
@@ -330,7 +330,7 @@ public String toDefaultValue(Schema p) {
     @Override
     public String toModelImport(String name) {
         // name looks like Cat
-        return "from " + modelPackage() + "." + toModelFilename(name) + " import "+ name;
+        return "from " + modelPackage() + "." + toModelFilename(name) + " import "+ toModelName(name);
```
Comment (spacether) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java:333]: Thank you for this fix!

This is definitely the right way to fix it.
Because our tests currently all pass, I am concerned that this corner case can break in the future and we would have no way to tell.

Can you add a test that verifies that the model class name is correct and the import is correct for a model named oddly? One could add this test in a couple locations:
- the [v3 sample tests](https://github.com/OpenAPITools/openapi-generator/tree/master/samples/openapi3/client/petstore/python-experimental/tests), how about adding a test of the model `_special_.model._name` which is named `SpecialModelName` in python
- the [java testing file](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonClientExperimentalTest.java) for this generator
